paxos/handlers.go

```diff
@@ -20,31 +16,27 @@ func (client *Client) HandlePETITION(message* Message) {
 func (client *Client) HandlePROPOSE(message *Message) {
   if message.GetBallot() > client.ballotNum {
     client.ballotNum = message.GetBallot()
-    client.Send(client.peers[message.GetNode()], &Message {
-      Type:   Message_PROMISE,
-      Epoch:  message.GetEpoch(),
-      Ballot: message.GetBallot(),
-      Value:  message.GetValue(),
-    })
+
+    reply := client.MakeReply(Message_PROMISE, message)
```

Comment:
I think we need to forward the acceptVal with highest acceptNum back to the leader